server/controllers/TripController.js

```diff
@@ -304,6 +304,40 @@ class TripController {
       throw new Error('Could not delete trip')
     }
   }
+
+  async enableShareTrip(userId, id) {
+    try {
+      await TripModel.updateOne(
+        { _id: id, userId: userId },
+        {
+          $set: {
+            isPublic: true, // update the field
+          },
+        },
+        function (err) {
+          if (err) throw err
+          console.log('1 document updated')
+        },
+      ).lean()
+    } catch (error) {
+      if (config.server.env === 'DEV')
+        console.error('Error while deleting trips:', error)
+      throw new Error('Could not enable sharing')
+    }
+  }
+
+  async getSharedTrip(id) {
+    try {
+      await TripModel.findOne({ _id: id, isPublic: true }, {}, function (err) {
+        if (err) throw err
+        console.log('retrieved shared trip')
+      }).lean()
+    } catch (error) {
+      if (config.server.env === 'DEV')
```

Comment:
same as above, we alr have error propagation to deal with logging and handling :3